GMG/UI/ViewModifier/CustomFontModifier.swift

```diff
@@ -0,0 +1,50 @@
+//  Copyright © 2025 ADA 4th GMG. All rights reserved.
+
+import SwiftUI
+
+struct LocalizedCustomFontModifier: ViewModifier {
+    @Environment(\.locale) private var locale: Locale
+
+    private let primary: LocalizedCustomFont
+    private let fonts: [LocalizedCustomFont]
+
+    init(primary: LocalizedCustomFont, secondary: [LocalizedCustomFont]) {
+        self.primary = primary
+        self.fonts = [primary] + secondary
+    }
+
+    private var fontToUse: CustomFont {
+        guard let currentCode: Locale.LanguageCode = locale.language.languageCode,
+            let localizedFont: LocalizedCustomFont = fonts.first(where: {
+                $0.languageCode == currentCode
+            })
+        else { return primary.customFont }
+        return localizedFont.customFont
+    }
```

Comment:
Local.LanguageCode랑 LocalizedCustomFont 타입을 꼭 지정하지 않아도 된다면 아래처럼 리펙토링해도 깔끔할 것 같아요


```suggestion
private var fontToUse: CustomFont {
    guard
        let code = locale.language.languageCode,
        let font = fonts.first(where: { $0.languageCode == code })
    else {
        return primary.customFont
    }

    return font.customFont
}
```

레몬이 적은 거랑 로직은 같은데 이름이 길다보니 코드 파악하기가 약간 어려웠어요!

Reply:
타입을 명시적으로 적으면 컴파일러 성능이 더 높아져서 가능하면 타입을 적으려고 하는데 길어져서 읽기가 어려울 것 같긴 하네요!